### PR TITLE
chore: Migrate codebase to Typescript

### PR DESCRIPTION
**Pull request by DominicKramer**
BREAKING CHANGE:  As a result of the migration, the way in which this package is required and used has changed slightly.

That is, the package is required and a `LoggingWinston` object is created as follows on Node 6+.  In particular, note the `{}` braces used in the require statement and the use of `new` which is now mandatory.

```javascript
// Imports the Google Cloud client library for Winston
const {LoggingWinston} = require('@google-cloud/logging-winston');

// Creates a Winston Stackdriver Logging client
const loggingWinston = new LoggingWinston();
```

On Node 4, the  require syntax is slightly different, but the use of `new` is still required.

```javascript
// Imports the Google Cloud client library for Winston
const LoggingWinston = require('@google-cloud/logging-winston').LoggingWinston;

// Creates a Winston Stackdriver Logging client
const loggingWinston = new LoggingWinston();
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -128,14 +128,14 @@ jobs:
       - run:
           name: Install modules and dependencies.
           command: |
-            npm install
+            npm install --unsafe-perm
```
Comment (ofrobots) [inline .circleci/config.yml]: Is this change still needed now that we have https://github.com/googleapis/nodejs-logging-winston/pull/25 landed.

File: .circleci/config.yml [hunk 2]
```diff
@@ -167,14 +167,14 @@ jobs:
       - run:
           name: Install and link the module.
           command: |
-            npm install
+            npm install --unsafe-perm
```
Comment (ofrobots) [inline .circleci/config.yml]: ditto

File: .circleci/config.yml [hunk 3]
```diff
@@ -201,7 +201,7 @@ jobs:
                 -k "${SYSTEM_TESTS_ENCRYPTION_KEY}"
       - run:
           name: Install modules and dependencies.
-          command: npm install
+          command: npm install --unsafe-perm
```
Comment (ofrobots) [inline .circleci/config.yml]: ditto

**Comment by ofrobots** [comment]
I would have expected a change to the `files` section of the package.json as well.

**Comment by DominicKramer** [comment]
Closed in favor of #27 that address merge conflicts.
